### PR TITLE
Threshold wallet

### PR DESCRIPTION
**Pull request by NhoxxKienn**
**Integration of Threshold BBS+ with VCWallet:**
The goal is to implement Threshold Wallet Client for Aries with the use of VCWallet package and Threshold BBS+ as signature scheme.

**Description:**
The primitive  implementation of threshold bbs+ can be found in `component/kmscrypto/crypto/primitive`, while the implementation of threshold wallet clients is in `pkg/client/vcwallet/thresholdwallet`.

The signer for the threshold bbs+ scheme is located in `component/models/signature/signer/threshold_bbs_signer.go`. The verifier for threshold bb+ can be reused from the normal bbs+ verifier.

The Threshold wallet clients: Holder and PartySigner are located in `pkg/client/vcwallet/thresholdwallet`. The package also contains a Precomputation Generator definition.

**Summary:**

- Implementation of threshold bbs+ signature schemes in Aries. (taking reference from `component/kmscrypto/crypto/primitive/bbs12381g2pub` and Go Implementation of Threshold BBS+ )
- Implementation of partial signature signers. ( taking reference from BBS Signer in `component/models/verifiable/credential_ldp_test.go`  and `BBSG2SignatureVerifier` in `component/models/signature/verifier/public_key_verifier.go` and the definition of signer and signature suite in `component/models/signature/signer` and `component/models/signature/suite` 
- Implementation of Threshold Wallet clients (Holder and Partial Signers) (taking reference from `pkg/client/vcwallet/client.go` )
- Interfaces and Implementation of Documents, Threshold Wallet Clients, Precomputation Generator.
- Simple Issue Credential Protocol Test (`pkg/client/vcwallet/thresholdwallet/issue_credential_test.go`).

**Future Changes**

- [x] Unit tests for threshold wallet clients.
- [ ] Fail cases for issue credential test.
- [ ] Partial signers unit test.
- [x] Documentation of implemented code.

### CODE REVIEW

File: component/models/signature/signer/threshold_bbs_signer.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+func (sv baseSignatureSigner) Algorithm() string {
+	return sv.algorithm
+}
+
+// ThresholdBBSG2SignaturePartySigner defines a party signer based on the baseSignatureSigner.
+// Party signer signs a credential and produces a partial signature based on its pregenerated presignature.
+type ThresholdBBSG2SignaturePartySigner struct {
+	partyPrivKeyBytes []byte
+	indices           [][]int
```
Comment (cryptphil) [inline component/models/signature/signer/threshold_bbs_signer.go]: Please add a comment here explaining this 2d array.

Comment (NhoxxKienn) [reply component/models/signature/signer/threshold_bbs_signer.go]: The commit 14e2dcdffca12f285d1fd857938da8ebb9c6888a solved this issue

**Comment by NhoxxKienn** [comment]
The commits adds DIDComm Services: didexchange and issuecredential for ThresholdWallet Clients. The implementation derived from their respective package in `pkg/client/didexchange` and `pkg/client/issuecredential` and `pkg/client/issuecredential/rfc0593`

- [update dependency](https://github.com/perun-network/aries-framework-go/pull/1/commits/79dd73c42fcb6435382723269588cbc02339321a) 

- [integrate didcomm issue credential for clients](https://github.com/perun-network/aries-framework-go/pull/1/commits/acfb1062cac20d923a929a93448c977880b5ed2c) 

- [turn off credential proof check to store partial signed credential](https://github.com/perun-network/aries-framework-go/pull/1/commits/1ebd3c484a50075d7299a3472c185f9fd77c6998) 

- [changes issue_threshold_credential_test with new DIDComm CLients](https://github.com/perun-network/aries-framework-go/pull/1/commits/241b9bdcec888a68c24ab6eaa85ee8a9e6cecebb) 

- [add comments for didcomm integration](https://github.com/perun-network/aries-framework-go/pull/1/commits/59a4b8c62f7d5298576e4c6065faac2333f76188)

**Comment by NhoxxKienn** [comment]
[fix module name for using in demo](https://github.com/perun-network/aries-framework-go/pull/1/commits/4283f01dab6749b48b74f9b3d94c704e4c2c8298) serves only as a workaround to use in the aries threshold credential issuance demo. Should be revert if merge to original

**Review by iljabvh — APPROVED** [review]

**Pull request merged**